drivers/serial/uart_realtek_rts5912.c

```diff
@@ -50,11 +134,84 @@ static int rts5912_uart_init(const struct device *dev)
 		return rc;
 	}
 
+#if defined(CONFIG_PM)
+	const uint32_t uart_reg = uart_ns16550_get_port(dev_cfg->uart_dev);
+	uint32_t iir = sys_read32(ADDR_IIR) & IIR_MASK;
+	uint32_t wf_cycle_count = k_us_to_cyc_ceil32(RTS5912_MAXIMUM_UART_POLLING_TIME_US);
+	uint32_t wf_start = k_cycle_get_32();
+	uint32_t wf_now = wf_start;
+
+	/*	For RTS5912 UART, if enable UART wake up function, it will
+	 *	change GPIO pin from uart funciton to gpio function before WFI.
+	 *	System need to ensure this register is cleared in every time doing init function.
+	 */
+	while (iir != IIR_NIP && (wf_cycle_count > (wf_now - wf_start))) {
```

Comment:
I recommend dropping the initialization of `iir` above and putting the only read of IIR inside the loop

```suggestion
	while (wf_cycle_count > (wf_now - wf_start)) {
		iir = sys_read32(ADDR_IIR) & IIR_MASK;
		if (iir == IIR_NIP) {
			break;
		}
		...
```